code/game/objects/explosion.dm

```diff
@@ -127,7 +127,7 @@
 		//Machines which report explosions.
 		if(!squelch)
 			for(var/obj/machinery/computer/bhangmeter/bhangmeter in doppler_arrays)
-				if(bhangmeter)
+				if(bhangmeter && !(bhangmeter.stat & BROKEN))

```

Comment:
Just to && !bhangmeter.stat, it's a better catch-all, since this doesn't count NOPOWER.